BangazonAPI/Controllers/DepartmentController.cs

```diff
@@ -0,0 +1,235 @@
+                    d.Name,
+                    d.Budget
+                    FROM Department d
+                    ";
+                using (IDbConnection conn = Connection)
+                {
+
+                    IEnumerable<Department> departments = await conn.QueryAsync<Department>(
+                        sql);
+
+                    return Ok(departments);
+                }
+            }
+            
+           
+
+           
+        }
+
+        // GET api/departments/5
+        [HttpGet("{id}", Name = "GetDepartment")]
+        public async Task<IActionResult> Get([FromRoute]int id)
+        {
+            string sql = $@"
+            SELECT
+                d.Id,
+                d.Name,
+                d.Budget
+            FROM Department d
+            WHERE d.Id = {id}
+            ";
+
+            using (IDbConnection conn = Connection)
+            {
+                IEnumerable<Department> departments = await conn.QueryAsync<Department>(sql);
+                return Ok(departments.Single());
+            }
+        }
+
+        // POST api/Department
+        [HttpPost]
+        public async Task<IActionResult> Post([FromBody] Department departments)
+        {
+            string sql = $@"INSERT INTO Department
+            (Name, Budget)
+            VALUES
+            (
+                '{departments.Name}',
+                '{departments.Budget}'
+            );
+            SELECT SCOPE_IDENTITY();";
+
+            using (IDbConnection conn = Connection)
+            {
+                var newId = (await conn.QueryAsync<int>(sql)).Single();
+                departments.Id = newId;
+                return CreatedAtRoute("GetDepartment", new { id = newId }, departments);
+            }
+        }
+
+
+        // PUT api/departments/5
+        [HttpPut("{id}")]
+        public async Task<IActionResult> Put(int id, [FromBody] Department departments)
+        {
+            string sql = $@"
+            UPDATE Department
+            SET Name = '{departments.Name}',
+            Budget = '{departments.Budget}'
+            WHERE Id = {id}";
+
+            try
+            {
+                using (IDbConnection conn = Connection)
+                {
+                    int rowsAffected = await conn.ExecuteAsync(sql);
+                    if (rowsAffected > 0)
+                    {
+                        return new StatusCodeResult(StatusCodes.Status204NoContent);
+                    }
+                    throw new Exception("No rows affected");
+                }
+            }
+            catch (Exception)
+            {
+                if (!DepartmentExists(id))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+        }
+
+        /*
+        // DELETE api/productTypes/5
+        [HttpDelete("{id}")]
+        public async Task<IActionResult> Delete(int id)
```

Comment:
Don't leave commented code in the file. If you don't need it, you don't need it...

Reply:
Doh just completely didn't see that. Lower down the page